### PR TITLE
Add example integration test with authentication in place.

### PR DESCRIPTION
**Pull request by felipecao**
Relates to https://github.com/rebuy-de/employee-tracking-silo/pull/8

@rebuy-de/prp-rebuy-silo-archetype @rebuy-de/prp-employee-tracking-silo

### CODE REVIEW

**Comment by ullrichp** [comment]
:+1:

File: src/main/resources/archetype-resources/silo/src/test/java/silo/configuration/TokenCacheTestConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import static org.mockito.Mockito.when;
+
+@Configuration
+@Profile("testing")
+public class TokenCacheTestConfiguration
+{
+    public final static UUID systemToken = UUID.randomUUID();
+
+    @Bean
+    public TokenCache tokenCache()
```
Comment (hendrik-rebuy) [inline src/main/resources/archetype-resources/silo/src/test/java/silo/configuration/TokenCacheTestConfiguration.java]: wouldn't this be much easier to handle by mocking the customer client create token

Comment (felipecao) [inline src/main/resources/archetype-resources/silo/src/test/java/silo/configuration/TokenCacheTestConfiguration.java]: I don't have a strong opinion on how to mock this, @hendrik-rebuy . If you could point to an example, I'd be more than happy to test it and put it into this PR :)

Comment (hendrik-rebuy) [inline src/main/resources/archetype-resources/silo/src/test/java/silo/configuration/TokenCacheTestConfiguration.java]: @Bean
    public CustomerClient customerClient()
    {
        CustomerClient mock = mock(CustomerClient.class);

        TokenDto tokenDto = new TokenDto();
        tokenDto.accessToken = CLIENT_TOKEN;
        tokenDto.expiresIn = TimeUnit.HOURS.toSeconds(1);

        when(mock.createToken()).thenReturn(tokenDto);

        return mock;
    }

Comment (felipecao) [inline src/main/resources/archetype-resources/silo/src/test/java/silo/configuration/TokenCacheTestConfiguration.java]: @hendrik-rebuy I've tried that and get a compilation error saying `Cannot resolve method 'createToken'`. I'm using customer-client 2.7.0.

Comment (hendrik-rebuy) [inline src/main/resources/archetype-resources/silo/src/test/java/silo/configuration/TokenCacheTestConfiguration.java]: sorry my bad its the permissionClient in the newest version

**Comment by felipecao** [comment]
looks better @hendrik-rebuy ?

**Comment by hendrik-rebuy** [comment]
:+1:

**Pull request merged**